### PR TITLE
Fix "keep me logged in"

### PR DESCRIPTION
**Pull request by bradycargle**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [X ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

When people log in, there is an option to "keep me logged in". This used to add a cookie to the browner named REMEMBERME, but this behavior broke somewhere down the line. Now people are getting logged out soon and having to log back in, often losing work.

Look at [this issue](https://github.com/mautic/mautic/issues/9804#issuecomment-1025756557) and [this issue](https://github.com/mautic/mautic/issues/9011#issuecomment-1009051271) for reference.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Before logging in, open the cookie viewer. You can do this by right click -> Inspect -> and go to Application. Scroll down to Storage -> Cookies and there you can see browser cookies
3. Log in and click "keep me logged in". You should see a cookie named REMEMBERME added to the browser cookies
4. Delete all cookies except the REMEMBERME cookie (click on the cookie, then Backspace). Reload the page. You should still be logged in
5. Delete all cookies (including REMEMBERME) and reload the page. You should be logged out

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10994"></a>

### CODE REVIEW

**Comment by ShareBtech** [comment]
I tried this on latest version of Mautic 4.2.0 and after copying file over to my /app/ directory and clearing cache I was unable to get to the site.

In the error log i get the following message:
`[error] 1851#1851: *784643 FastCGI sent in stderr: "PHP message: ParseError: syntax error, unexpected ''lifetime'' (T_CONSTANT_ENCAPSED_STRING), expecting ']' - in file /var/www/mautic/app/config/security.php - at line 98" while reading response header from upstream, client: 167.83.147.45, server: mymautic.com,

**Comment by bradycargle** [comment]
@ShareBtech Might be a few ways to fix this. It sounds like the file might have been copied to the wrong directory; can you make sure the change is only in app/config/security.php?

If that's not the problem, let's try this: Revert back to your Mautic instance from before the change, then open your console, cd to the Mautic folder, and type `gh pr checkout 10994`. This will make the exact file changes that's in my PR. See if that works for you

Last, we can always test with GitPod. You can click on the "Open in GitPod" button above. It's working for me there. I found the cookie using Right Click > Inspect > Application > Cookies > https://8080-bradycargle...

File: app/bundles/CoreBundle/Tests/Functional/RememberMe/RememberMeTest.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Mautic\CoreBundle\Tests\Functional;
```
Comment (escopecz) [inline app/bundles/CoreBundle/Tests/Functional/RememberMe/RememberMeTest.php:3]:
```suggestion
namespace Mautic\CoreBundle\Tests\Functional\Config;
```
Can we rename the new folder to Config so more tests regarding config issues could be add in it?

File: app/config/security.php [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
 
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
```
Comment (escopecz) [inline app/config/security.php:11]:
```suggestion
```
These doc blocks are no longer needed

**Comment by escopecz** [comment]
Why was this closed? I don't see who closed it. Reopening.

**Comment by adiux** [comment]
@escopecz good catch! @bradycargle could you fix the CS Fixer problem?

**Comment by jos0405** [comment]
> @escopecz good catch! @bradycargle could you fix the CS Fixer problem?

@bradycargle probably won't work on this one, is anyone else who can fix the CS?
@escopecz @adiux @kuzmany ?

**Comment by luizeof** [comment]
Hi @RCheesley, any chance this pull will be part of the 4.4 release?

**Comment by RCheesley** [comment]
@luizeof no, the release is already finalised and this PR still requires work, as there is a failing test, and will need to be tested by community members before it can be merged.

If you had this in production, please use the review feature under the 'files changed' at the top of the page to leave a review.

**Comment by biozshock** [comment]
@escopecz @RCheesley i have a solution for the root cause of the parameter issue. See #11363 .

**Comment by escopecz** [comment]
Closing in favor of https://github.com/mautic/mautic/pull/11363

**Pull request closed without merging**